### PR TITLE
Remove experimental namespace from SYCL

### PR DESCRIPTION
**Pull request by Parsifal-2045**
This is a supporting PR for #1845 
It only removes all the experimental namespaces related to SYCL. Memory operations are currently not working (the bufCopy example won't even compile). That is being fixed by the main PR but seems to be related to the importing of mdspan.

### CODE REVIEW

File: include/alpaka/mem/view/Accessor.hpp
Comment (psychocoderHPC) [inline include/alpaka/mem/view/Accessor.hpp]: Please check that you move only SYCL stuff out of the experimental namespace.

File: include/alpaka/mem/view/ViewAccessOps.hpp
Comment (psychocoderHPC) [inline include/alpaka/mem/view/ViewAccessOps.hpp]: should be reverted, SYCL independent.

File: include/alpaka/mem/view/ViewAccessor.hpp
Comment (psychocoderHPC) [inline include/alpaka/mem/view/ViewAccessor.hpp]: SYCL independent

**Review by psychocoderHPC — DISMISSED** [review]

File: include/alpaka/mem/fence/MemFenceGenericSycl.hpp
Comment (j-stephan) [inline include/alpaka/mem/fence/MemFenceGenericSycl.hpp:62]: Since you are touching this region anyway, could you remove the `ext::oneapi` namespace here, too?

**Review by j-stephan — CHANGES_REQUESTED** [review]
This looks mostly good. @psychocoderHPC has already addressed most of the issues that need to be fixed. Also, please update the copyright header in all files you touched here. Additionally:

File: include/alpaka/mem/fence/MemFenceGenericSycl.hpp
Comment (Parsifal-2045) [reply include/alpaka/mem/fence/MemFenceGenericSycl.hpp:62]: The last one is an easy fix. However, all the other changes are more involved as SYCL Accessor relies on the three other mentioned files. If I revert changes on those, most of the SYCL accessor will have to go back in the experimental namespace.

File: include/alpaka/mem/view/Accessor.hpp
Comment (Parsifal-2045) [reply include/alpaka/mem/view/Accessor.hpp]: As far as I can tell, this Accessor file is included only in SYCL (buffer and accessor). I have not found a way to leave these files inside the experimental namespace without breaking SYCL

Comment (bernhardmgruber) [reply include/alpaka/mem/view/Accessor.hpp]: Yes, I built the accessors a long time ago to prepare for the SYCL backend. But when we move to pointer based SYCL, they are no longer needed. Except for the use cases of @psychocoderHPC who wants to configure how memory access works, but this can also be handled by `std::mdspan`. I want to drop accessors in a separate PR, see also #1897.

**Comment by bernhardmgruber** [comment]
I rebased the PR and fixed @psychocoderHPC's suggestions.

**Comment by bernhardmgruber** [comment]
All examples compile with `dpcpp 2023.0.0`, the tests fail to compile with lots of errors unrelated to this PR.

**Comment by bernhardmgruber** [comment]
@Parsifal-2045 do you want to make more changes to this PR or can we merge it? (The PR is in draft mode)

**Review by bernhardmgruber — DISMISSED** [review]

**Comment by Parsifal-2045** [comment]
> @Parsifal-2045 do you want to make more changes to this PR or can we merge it? (The PR is in draft mode)

I'll just modify the headers and remove the ext::oneapi as suggested, in the next few hours. I'll update the PR once it's ready to be merged

**Review by bernhardmgruber — APPROVED** [review]

**Review by j-stephan — APPROVED** [review]
Thanks!

**Comment by j-stephan** [comment]
GitHub has some server issues at the moment. We'll hit the merge button later once they are resolved.

**Pull request merged**
